include/core/interface_tools.h

```diff
@@ -286,6 +304,414 @@ namespace InterfaceTools
+     *
+     * @param[in] mpi_communicator MPI communicator
+     */
+    void
+    set_level_set_from_background_mesh(
+      const DoFHandler<dim> &background_dof_handler,
+      const VectorType      &background_level_set_vector,
+      const MPI_Comm        &mpi_communicator);
+
+    /**
+     * @brief solve
+     *
+     * Solve for the signed distance from the given level of the level-set
+     * vector.
+     */
+    void
+    solve(const MPI_Comm &mpi_communicator);
+
+    /**
+     * @brief Getter methods to get the private attribute signed_distance
+     *
+     * @param[in] mpi_communicator MPI communicator
+     *
+     * @return vector storing the computed signed distance
+     */
+    VectorType &
+    get_signed_distance(const MPI_Comm &mpi_communicator);
+
+    /**
+     * @brief Output the interface reconstruction used for the signed distance
+     * computations
+     *
+     * @param[in] output_name name of the output file
+     *
+     * @param[in] output_path path to the output file
+     *
+     * @param[in] it iteration number
+     *
+     */
+    void
+    output_interface_reconstruction(const std::string  output_name,
+                                    const std::string  output_path,
+                                    const unsigned int it) const;
+
+    /**
+     * @brief Attach the solution vector to the DataOut provided. This function
+     * enable the auxiliary physics to output their solution via the background
+     * solver.
+     *
+     * @param[in,out] data_out DataOut reponsible for solution output
+     */
+    void
+    attach_solution_to_output(DataOut<dim> &data_out);
+
+    /// DoFHandler describing the problem
+    DoFHandler<dim> dof_handler;
+
+  private:
+    /**
+     * @brief Zero the ghost dofs entries of the solution vectors to gain write
+     * access to the ghost elements
+     */
+    void
+    zero_out_ghost_values();
+
+    /**
+     * @brief Update the ghost dofs entries of the solution vectors to gain read
+     * access to the ghost elements
+     */
+    void
+    update_ghost_values();
+
+    /**
+     * @brief Exchange the distance and set it to the minimum value across the
+     * processors by using the compress(VectorOperation::min) function
+     */
+    void
+    exchange_distance();
+
+    /**
+     * @brief Initialize the distance solution vectors to the given max_distance
+     */
+    void
+    initialize_distance();
+
+    /**
+     * @brief Compute the geometric distance (brute force) between the interface
+     * reconstruction and the dofs of the intersected cells (first neighbors)
+     */
+    void
+    compute_first_neighbors_distance();
+
+    /**
+     * @brief Compute the geometric distance (marching method) between the
+     * interface reconstruction and the rest of the dofs (second neighbors)
+     *
+     * @param[in] mpi_communicator MPI communicator
+     */
+    void
+    compute_second_neighbors_distance(const MPI_Comm &mpi_communicator);
```

Comment:
Do you need to pass the communicator here since this is a member function of the class and the class has the dof handler?
You can always get the communicator from the dof handler (see https://www.dealii.org/current/doxygen/deal.II/classDoFHandler.html#acd999cd3bf8d91a1157d16b01872664c)

Reply:
I changed it everywhere!